manifest.go

```diff
@@ -156,6 +156,29 @@ func validateManifest(s string) ([]error, error) {
 	return warns, nil
 }
 
+// ValidateProjectRoots validates the project roots present in manifest.
+func ValidateProjectRoots(c *Ctx, m *Manifest, sm gps.SourceManager) error {
+	projectRoots := make([]gps.ProjectRoot, 0, len(m.Constraints)+len(m.Ovr))
+	for pr := range m.Constraints {
+		projectRoots = append(projectRoots, pr)
+	}
+	for pr := range m.Ovr {
+		projectRoots = append(projectRoots, pr)
+	}
+
+	for _, pr := range projectRoots {
```

Comment:
let's do these concurrently, otherwise any that require http requests for vanity URLs will end up being serialized.